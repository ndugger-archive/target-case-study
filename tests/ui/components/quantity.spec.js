import React from 'react';
import { shallow } from 'enzyme';

import Quantity from '../../../src/ui/components/quantity';

test('quantity -> should match snapshot', () => {
    const quantity = shallow(<Quantity/>);

    expect(quantity).toMatchSnapshot();
});

test('quantity.state.value -> initial value should be 1', () => {
    const quantity = shallow(<Quantity/>);
    const instance = quantity.instance();

    expect(instance.state.value).toEqual(1);
});

test('quantity.handleSubtractQuantity -> should NOT subtract any value if the value is 1', () => {
    const quantity = shallow(<Quantity/>);
    const instance = quantity.instance();

    instance.handleSubtractQuantity();

    expect(instance.state.value).toEqual(1);
});

test('quantity.handleSubtractQuantity -> should subtract 1 from quantity.state.value', () => {
    const quantity = shallow(<Quantity/>);
    const instance = quantity.instance();

    instance.setState({ value: 2 });
    instance.handleSubtractQuantity();

    expect(instance.state.value).toEqual(1);
});

test('quantity.handleAddQuantity -> should add 1 to quantity.state.value', () => {
    const quantity = shallow(<Quantity/>);
    const instance = quantity.instance();

    instance.handleAddQuantity();

    expect(instance.state.value).toEqual(2);
});
