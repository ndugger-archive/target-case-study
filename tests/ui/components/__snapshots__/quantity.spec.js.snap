exports[`test quantity -> should match snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.section>
    <styled.header>
        quanitity:
    </styled.header>
    <styled.div>
        <Styled(Icon)
            glyph="remove_circle"
            onClick={[Function]} />
        <styled.span>
            1
        </styled.span>
        <Styled(Icon)
            glyph="add_circle"
            onClick={[Function]} />
    </styled.div>
    <styled.div />
</styled.section>,
  "nodes": Array [
    <styled.section>
      <styled.header>
            quanitity:
      </styled.header>
      <styled.div>
            <Styled(Icon)
                  glyph="remove_circle"
                  onClick={[Function]} />
            <styled.span>
                  1
            </styled.span>
            <Styled(Icon)
                  glyph="add_circle"
                  onClick={[Function]} />
      </styled.div>
      <styled.div />
</styled.section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Quantity />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Quantity {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "value": 1,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.section>
          <styled.header>
                    quanitity:
          </styled.header>
          <styled.div>
                    <Styled(Icon)
                              glyph="remove_circle"
                              onClick={[Function]} />
                    <styled.span>
                              1
                    </styled.span>
                    <Styled(Icon)
                              glyph="add_circle"
                              onClick={[Function]} />
          </styled.div>
          <styled.div />
</styled.section>,
        "_debugID": 2,
        "_renderedOutput": <styled.section>
          <styled.header>
                    quanitity:
          </styled.header>
          <styled.div>
                    <Styled(Icon)
                              glyph="remove_circle"
                              onClick={[Function]} />
                    <styled.span>
                              1
                    </styled.span>
                    <Styled(Icon)
                              glyph="add_circle"
                              onClick={[Function]} />
          </styled.div>
          <styled.div />
</styled.section>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Quantity />,
}
`;
